src-electron/importexport/file-format.js

```diff
@@ -0,0 +1,310 @@
+/**
+ *
+ *    Copyright (c) 2023 Silicon Labs
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
```

Comment:
The name of this file suggests that it formats the older longer ZAP file into the newer shorter version. 

As far as I can tell it takes the new version and reads it so the database does not need to be changed. 

Maybe "file-interpreter.js" or "file-translator.js" ?

Reply:
It does both. It performs both input and output translation of the shorter version. The "longer" version, is the canonical version, so this is just post- and pre-processing.